docs/CHANGELOG.rst

```diff
@@ -32,12 +32,12 @@ Main new features
 - Run time display color format adjustment with RGB888 support
 - Built-in support ``pthread``, ``FreeRTOS`` and other (RT)OSes which are used during rendering
 - Built-in support LVGL's, C library, and other stdlib's
-- Better parallel rendering architecture. See the details :ref:`here <porting/draw/custom gpu [0-1]>`
+- Better parallel rendering architecture. See the details :ref:`here <porting/draw/custom gpu>`
 - Built in display and touch driver: SDL, Linux Frame buffer,  NuttX LCD and touch drivers, ST7789 and ILI9341 driver are available and more will come soon
-- :ref:`others/observer/observer [0-1]` allows to bind data to UI elements and create a uniform and easy to maintain API
+- :ref:`others/observer/index/observer` allows to bind data to UI elements and create a uniform and easy to maintain API
```

Comment:
Why do we have `index` here?

Reply:
It's to correct the duplicate "label" issue. The section names should remain the same as they are. If you read the documentation you will understand why they do. In order to handle that the observer documentation has been broken up into several files so the section names are not duplicates. This is because they are in different files.

Look at the file/folder structure in the docs and it will make more sense.

Reply:
I modified the document builder code that inserts the API documentation so it will now read index files. This allows for nesting TOC trees into documentation where there is an API section after a TOC tree directive. I use the folder name as the name to find any matching header files when dealing with index.rst files.

Reply:
Only the "Observer" section was duplicated, right?

Reply:
I should be able to remove the "index" from the references if you would like. I don't think that any name collisions would occur if I do.

Reply:
No there were a couple of other locations that also had duplicated sections. If it was only the one I would have renamed the headers. This is why I thought it would be best to go the route I went, It is also not just one section in a file that is bumping heads. some of the files have 3-4 sections that had naming conflicts.  

again I can remove the need to have "index" in the ref directive if you like.

Reply:
Is it possible to rollback this `observer` to store all the observer doc in 1 file?